### DISCUSSION TITLE
Pre-release picks tag for wrong release-channel

### DISCUSSION THREAD

**Discussion by Tlepel** [semantic-release/semantic-release#2414] [Q&A] [answered]
Hi everyone!

In my project there are 3 branches: `production`, `development` and `beta`. We've only recently started releasing on the alpha branch, in addition to the releases on the beta and production branches where we've been releasing for a few years now.

We've released a version to the alpha branch, let's say `v2.2.0-development.3`, which worked perfectly fine. However, when all seemed well we merged the alpha branch to beta and tried to release there. We don't use merge commits for this, so the `v2.2.0-beta.1` tag was created on the same commit as the `v2.2.0-development.3` tag. Now we've pushed a few commits to the beta branch that fixed some issues on that branch, and tried to release that. However, the tag that was generated was `v2.2.0-development.4` instead of `v2.2.0-beta.2`..

I checked the git notes to see if it went wrong there, and I tried to adapt it to reflect the situation, but that didn't change the version at all; none of these options worked:

```
{"channels": ["development","beta"]}
{"channels": ["beta"]}
{"channels": ["development"]}
```

### Schematic:

```
|
* - v2.2.0-development.1
|
* - v2.2.0-development.2
|
* - v2.2.0-development.3, v2.2.0-beta.1
| \
*  \   - v2.2.0-development.4 (on development branch)
|   |
|   *  - v2.2.0-development.4 (on beta branch, should be v.2.2.0-beta.2)
```

### Setup:

```js
branches: [
  'production',
  {
    name: 'beta',
    prerelease: true,
  },
  {
    name: 'development',
    prerelease: true,
  },
],
```

Can anyone help me out and see what we're doing wrong here?

**Comment by Tlepel** [comment, marked as answer]
I created issue #2415 as the problem described above appeared to be a design limitation of the current implementation, and subsequently opened PR #2416, which will fix the issue.

Edit: the PR has been merged and the issue fixed

**Reply by shiipou** [reply]
Did you know a workaround to use until the PR got merged ?
Thanks for you work, great job !

**Reply by Tlepel** [reply]
We use [`patch-package`](https://www.npmjs.com/package/patch-package) for this until the fix has been merged. That might be a suitable solution for you as well?

**Reply by shiipou** [reply]
Thanks I'll look at it
